src/catalogue/category/property/catalogueItemPropertiesTable.component.tsx

```diff
@@ -173,7 +173,10 @@ export function CatalogueItemsPropertiesTable(props: PropertiesTableProps) {
       {
         header: 'Unit',
         Header: TableHeaderOverflowTip,
-        accessorFn: (row) => row.unit,
+        accessorFn: (row) =>
+          requestType === 'patch'
+            ? row.unit
+            : (units?.find((unit) => row.unit_id === unit.id) || null)?.value,
```

Comment:
Why is this different between patch and post? I thought they `row.unit` was always string unit value?

Reply:
Good spot the implementation is broken for patch. it need also needs the the second option as the unit_id is only thing store when sending a post request.